provider/source/src/currency/compact.rs

```diff
@@ -91,14 +91,33 @@ impl DataProvider<ShortCurrencyCompactV1Marker> for SourceDataProvider {
 
             let count = try_parse_count_from_str(pattern.count.as_str())?;
 
-            result.insert(&(lg10, count), pattern.pattern.as_str());
+            result.insert((lg10, count), pattern.pattern.as_str());
         }
 
+        // Deduplicate against `::Other`
+        let compact_patterns = result
+            .iter()
+            .filter(|(&(lg10, count), pattern)| {
+                let (CompactCount::AlphaNextToNumber(p) | CompactCount::Standard(p)) = count;
```

Comment:
nit: please explain in a comment how all of these checks contrubite to a dedup

Reply:
I'm unsure what you want me to explain. There's already a comment saying that I'm deduplicating against the other case.